src/ode_solver/ode_solver_base.cpp

```diff
@@ -214,9 +215,12 @@ int ODESolverBase<dim,real,MeshType>::advance_solution_time (double time_advance
     pcout
             << " Advancing solution by " << time_advance << " time units, using "
             << number_of_time_steps << " iterations of size dt=" << constant_time_step << " ... " << std::endl;
-    allocate_ode_system ();
+    if(!this->all_parameters->use_energy)
+        this->current_iteration = 0;
```

Comment:
@jbrillon , can you confirm that if somehow that flag isn't turned on, that we should reset iteration count?

Reply:
From what I saw, Julien doesn't use the function advance_solution_time for the flow solver. Instead, he calls step_in_time directly and does his own allocation/iteration loop.
From line 219 in https://github.com/dougshidong/PHiLiP/blob/master/src/ode_solver/ode_solver_base.cpp the current iteration is set to 0 there. The flag is because my explicit tests do a time loop in the testing directory.

Reply:
So tests should depend on the code, not the other way around. Someone should be able to use advance solution time independently of the knowledge of "energy" tests. I see this flag being used in the code to modify behavior of the code for the sake of your test.

I would say that removing this flag throughout the code that modifies behavior by replacing with different functions through either polymorphism (by having various derived class do different things rather than an if-statement) is definitely in the list of to-dos.

Reply:
Yes I agree entirely, it was a hard coded short cut.

Reply:
Will be resolved by issue #154